org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java

```diff
@@ -463,6 +467,16 @@ public void projectConfigurationUpdate(TextDocumentIdentifier param) {
 		return handler.resolveClasspaths(param);
 	}
 
+	/* (non-Javadoc)
+	 * @see org.eclipse.jdt.ls.core.internal.JavaProtocolExtensions#buildWorkspace(java.lang.String)
+	 */
+	@Override
+	public CompletableFuture<Object> buildWorkspace(String type) throws CoreException {
```

Comment:
from the implementation of [lsp4j](https://github.com/eclipse/lsp4j/blob/ffd31df92dcc91690d34c5f563754cdd2cb6c70a/org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/RemoteEndpoint.java), it will mark it as server error. i tested locally, ls won't exit. but i think you are right, return a build status is a better choice